src/python-common/ceph/deployment/service_spec.py

```diff
@@ -728,18 +728,17 @@ def __init__(self,
         if not self.api_secure and self.ssl_cert and self.ssl_key:
             self.api_secure = True
 
+        if not self.api_user:
+            self.api_user = "admin"
+        if not self.api_password:
+            self.api_password = "admin"
+
     def validate(self):
         super(IscsiServiceSpec, self).validate()
 
         if not self.pool:
             raise ServiceSpecValidationError(
                 'Cannot add ISCSI: No Pool specified')
-        if not self.api_user:
-            raise ServiceSpecValidationError(
-                'Cannot add ISCSI: No Api user specified')
-        if not self.api_password:
-            raise ServiceSpecValidationError(
-                'Cannot add ISCSI: No Api password specified')
```

Comment:
Do we need to upgrade existing iscsi specs? If no, can you please add a comment that you added those checks and that we don't need to handle this case? 

Note that we don't have a test for a iSCSI p2p upgrade. which means there is no validation that this actually upgrades properly in Teuthology

Reply:
No, I think we probably should handle upgrades. I could change this default to occur during the config_dashboard (the same way it is done for the api port) and then that wouldn't be an issue?

Reply:
Actually, I don't think anything would have to be done with an old iSCSI spec as they would have to have api_user and api_password set and the changes related to having defaults for those fields would have no effect. I added a comment saying we don't need to check for those fields because they're given default values. Is that what you were looking for or am I misunderstanding your comment? @sebastian-philipp

Reply:
yeah, just put this comment ^^^ into the docs.